lib/eventFactory.js

```diff
@@ -473,53 +474,100 @@ function updateWorkflowGraph(config) {
 }
 
 /**
- * Standardize event parameters
- * @method updateEventParameters
+ * Determines the parameters for the build by compiling the list of parameters with default values from the
+ * `defaultParameters` and override the values if specified in `eventParameters`. Parameters from `eventParameters` are
+ * dropped if matching definition is not found in `defaultParameters`.
+ * @param {Object} defaultParameters Default Pipeline or Job parameter definitions
+ * @param {Object} eventParameters   Customized build parameters
+ * @resolves {Object}
+ */
+function validateAndMergeParameters(defaultParameters, eventParameters) {
+    const allowedParameters = Object.create(null);
```

Comment:
wondering why we're using `Object.create(null)` vs `{}`, but I see that it was used in updateEventParameters() already